pkg/deployment/inventory.go

```diff
@@ -73,106 +134,30 @@ func GenerateRoleInventory(ctx context.Context, helper *helper.Helper,
 
 	invData, err := inventory.MarshalYAML()
 	if err != nil {
-		utils.LogErrorForObject(helper, err, "Could not parse Role inventory", instance)
+		utils.LogErrorForObject(helper, err, "Could not parse NodeSet inventory", instance)
 		return "", err
 	}
 	cmData := map[string]string{
 		"inventory": string(invData),
 		"network":   string(instance.Spec.NodeTemplate.NetworkConfig.Template),
 	}
-	configMapName := fmt.Sprintf("dataplanerole-%s", instance.Name)
-	cms := []utils.Template{
-		// ConfigMap
-		{
-			Name:         configMapName,
-			Namespace:    instance.Namespace,
-			Type:         utils.TemplateTypeNone,
-			InstanceType: instance.Kind,
-			CustomData:   cmData,
-			Labels:       instance.ObjectMeta.Labels,
-		},
-	}
-	err = configmap.EnsureConfigMaps(ctx, helper, instance, cms, nil)
-	return configMapName, err
-}
 
-// GenerateNodeInventory yields a parsed Inventory for node
-func GenerateNodeInventory(ctx context.Context, helper *helper.Helper,
-	instance *dataplanev1.OpenStackDataPlaneNode,
-	instanceRole *dataplanev1.OpenStackDataPlaneRole) (string, error) {
-	var (
-		err      error
-		hostName string
-	)
-
-	inventory := ansible.MakeInventory()
-	all := inventory.AddGroup("all")
-	host := all.AddHost(instance.Name)
-
-	ipSet := &infranetworkv1.IPSet{}
-	err = helper.GetClient().Get(ctx,
-		types.NamespacedName{Name: instance.Name, Namespace: instance.Namespace}, ipSet)
-	if err != nil {
-		if !k8s_errors.IsNotFound(err) {
-			return "", err
-		}
-		// Don't try to popluare inventory
-		utils.LogForObject(helper, "Networks not configured for Role", instance)
-	} else {
-		dnsAddresses, _, _, err := checkDNSService(ctx, helper, instance)
-		if err != nil {
-			return "", err
+	var configMapName string
+	for _, configMapName = range configMaps {
+		cms := []utils.Template{
+			// ConfigMap
+			{
+				Name:         configMapName,
+				Namespace:    instance.Namespace,
+				Type:         utils.TemplateTypeNone,
+				InstanceType: instance.Kind,
+				CustomData:   cmData,
```

Comment:
This doesn't seem right. It looks like all these configmaps will have the same data. I think we can probably just remove all of the per node COnfigMap inventory generation.

Reply:
Yeah, way back in the dawn of this PR we discussed this topic. I think it might have been Nova that used the individual node CM inventory. So now that we have removed Nova stuff, we can probably drop the individual node CM generation.